types/datum.go

```diff
@@ -880,7 +880,12 @@ func (d *Datum) convertToFloat(sc *stmtctx.StatementContext, target *FieldType)
 		err = err1
 	}
 	if target.Tp == mysql.TypeFloat {
-		ret.SetFloat32(float32(f))
+		if f > math.MaxFloat32 {
```

Comment:
Do we have to check `f < -math.MaxFloat32` here?

Reply:
Thx.